pkg/tracegen/templated.go

```diff
@@ -105,6 +107,8 @@ type Link struct {
 type Event struct {
 	// Name of event
 	Name string `js:"name"`
+	// PercentageOfSpan if set controls where in the span the event is generated. If missing, the event is generated randomly
+	PercentageOfSpan float32 `js:"percentageOfSpan"`
```

Comment:
I see a couple of potential improvements here:

1. It might be more consistent if the default behavior always adds the event when `PercentageOfSpan` is not set
2. What do you think about renaming it to `Rate` or `EventRate` for clarity?
3. Unless I'm missing something, the current implementation seems to use PercentageOfSpan only to calculate the event time (L322) but does not influence whether the event is actually added